src/actors/TableActor.js

```diff
@@ -0,0 +1,22 @@
+import * as SearchClient from "../clients/SearchClient";
+
+export class TableActor {
+  constructor(filterBarStore, tableStore) {
+    this.filterBarStore = filterBarStore;
+    this.tableStore = tableStore;
+  }
+
+  fetchData(page) {
+    if (page !== undefined) {
+      this.tableStore.setCurrentPage(page);
+    }
+
+    var url = this.tableStore.getUrl();
+    SearchClient.search(url, this.tableStore.updateTable.bind(this.tableStore));
```

Comment:
A better place for the url to be stored would be in the client itself.

Also update the currentPage when the search results are returned. I can see race conditions appearing here.